invenio/templates/configurations/logstash.yaml

```diff
@@ -53,66 +53,125 @@ data:
+        # Enrich with geolocation data.
+        geoip {
+            source => "http_x_forwarded_for" #field name that contains the ip address
+            fields => [ "city_name", "country_name", "country_code2", "continent_code"]
+            target => "data"
+            # Download static database and point to it
         }
-        # add ua/patch only if set
-        if [@metadata][ua_patch] {
-            mutate {
-                add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}.%{[@metadata][ua_patch]}" }
-            }
-        } else {
+
+      useragent {
+          source => "[data][user_agent]"
+          target => "[@metadata]"
+      }
+      ##############################################################
+      # USER AGENT
+      ##############################################################
+      # add ua/device only if set
+      if [@metadata][ua_device] {
             mutate {
-                add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}" }
+                add_field => { "[data][ua/device]" => "%{[@metadata][ua_device]}" }
             }
         }
-        # add ua/os only if set
-        if [@metadata][ua_os] {
-            mutate {
-                add_field => { "[data][ua/os]" => "%{[@metadata][ua_os]}" }
-            }
+      # add ua/family only if set
+      if [@metadata][ua_name] {
+          mutate {
+              add_field => { "[data][ua/family]" => "%{[@metadata][ua_name]}" }
+          }
+      }
+      # add ua/patch only if set
+      if [@metadata][ua_patch] {
+          mutate {
+              add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}.%{[@metadata][ua_patch]}" }
+          }
+      } else {
+          mutate {
+              add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}" }
+          }
+      }
+      # add ua/os only if set
+      if [@metadata][ua_os] {
+          mutate {
+              add_field => { "[data][ua/os]" => "%{[@metadata][ua_os]}" }
+          }
+      }
+      # add a field with the category, based on the url
+      grok {
+         match => { "url" => ["^/(api/)?record/%{INT}/files/", "^/api/files/%{UUID}/", "^/api/deposit/depositions/%{INT}/files"]}
+         add_field => {"[data][request_category]" => "files" }
+      }
+      if "_grokparsefailure" in [tags] {
+        grok {
+           match => { "url" => "^/%{WORD:[data][request_category]}"}
+           remove_tag => "_grokparsefailure"
         }
+     }
+     if [data][request_category] not in [ "files", "api", "oai2d", "static" ]  {
+        mutate { replace => {"[data][request_category]" => "ui"}}
+     }
+
+    {{ if .Values.logstash.cern_monit }}
+    mutate { rename => { "data" => "[m][data]" }}
+    mutate {
+      add_field => {"[m][producer]" => "zenodo"
```

Comment:
This line (and the next) should probably come from the values. The part that I don't know is if it should be a new value within the logstash part, or if we can reuse it from other places

Reply:
I would go for:
```
.Values.logstash.cern_monit.producer
.Values.logstash.cern_monit.type
```
environment, do we have to put `rdm` in front?

Reply:
Thanks, I'll change the producer and type.

For the environment, at least for zenodo, I wanted to differentiate between the current `production` and the `rdmproduction`